### PR TITLE
Added update call when switching between pages

### PR DESCRIPTION
**Pull request by laurens-ritense**
### Describe the changes
Link to the related Github issue:

**Relevant comments:**
>

### Breaking changes

- [x] The contribution only contains changes that are not breaking.

### Documentation

- [x] Release notes have been written for these changes.

Link to the pull request in the [Valtimo documentation repository](https://github.com/valtimo-platform/valtimo-documentation):
>

New features or changes that have been introduced have been documented.
- [x] Yes
- [ ] Not applicable

### CODE REVIEW

File: projects/valtimo/form-view-model/src/lib/components/form-view-model/form-view-model.component.ts [hunk 1]
```diff
@@ -79,6 +79,9 @@ export class FormViewModelComponent implements OnInit {
 
   public errors: string[] = [];
 
+  private preventNextPage = false;
+  private preventPreviousPage = false;
```
Comment (teodor-ritense) [inline projects/valtimo/form-view-model/src/lib/components/form-view-model/form-view-model.component.ts]:
```suggestion
  private _preventNextPage = false;
  private _preventPreviousPage = false;
```

File: projects/valtimo/form-view-model/src/lib/components/form-view-model/form-view-model.component.ts [hunk 2]
```diff
@@ -237,6 +240,28 @@ export class FormViewModelComponent implements OnInit {
     }
   }
 
+  public onNextPage(event: any): void {
+    this._preventNextPage = true;
+    this.formio.formio.setPage(this.formio.formio.page - 1);
+    this.handleChanges();
+  }
+
+  public onPreviousPage(event: any): void {
```
Comment (teodor-ritense) [inline projects/valtimo/form-view-model/src/lib/components/form-view-model/form-view-model.component.ts]:
```suggestion
  public onNextPage(): void {
    this._preventNextPage = true;
    this.formio.formio.setPage(this.formio.formio.page - 1);
    this.handleChanges();
  }

  public onPreviousPage(): void {
```

File: projects/valtimo/form-view-model/src/lib/components/form-view-model/form-view-model.component.html [hunk 3]
```diff
@@ -46,5 +46,7 @@
     (submit)="onSubmit($event)"
     (change)="onChange($event)"
     (focusout)="onBlur($event)"
+    (nextPage)="onNextPage($event)"
+    (prevPage)="onPreviousPage($event)"
```
Comment (teodor-ritense) [inline projects/valtimo/form-view-model/src/lib/components/form-view-model/form-view-model.component.html]:
```suggestion
    (nextPage)="onNextPage()"
    (prevPage)="onPreviousPage()"
```

**Review by teodor-ritense — DISMISSED** [review]

**Comment by laurens-ritense** [comment]
Don't merge before https://github.com/valtimo-platform/valtimo-backend-libraries/pull/1638

**Review by mbritense — DISMISSED** [review]

File: projects/valtimo/form-view-model/src/lib/services/view-model.service.ts [hunk 4]
```diff
@@ -83,13 +90,20 @@ export class ViewModelService extends BaseApiService {
...
+    isWizard: boolean
   ): Observable<object> {
+    const params = {
+      formName,
+      processDefinitionKey,
+      isWizard
+    }
+    if (!isNaN(page)) {
+      params['page'] = page;
+    }
```
Comment (teodor-ritense) [inline projects/valtimo/form-view-model/src/lib/services/view-model.service.ts]:
```suggestion
    const params = {
      formName,
      processDefinitionKey,
      isWizard,
      ...(!isNaN(page) && {page})
    }
```

File: projects/valtimo/form-view-model/src/lib/services/view-model.service.ts [hunk 5]
```diff
@@ -83,13 +90,20 @@ export class ViewModelService extends BaseApiService {
...
     return this.httpClient.post(this.getApiUrl(`/v1/form/view-model/start-form`), viewModel, {
-      params: {
-        formName,
-        processDefinitionKey,
-      },
+      params: params,
```
Comment (teodor-ritense) [inline projects/valtimo/form-view-model/src/lib/services/view-model.service.ts]:
```suggestion
      params,
```

File: projects/valtimo/form-view-model/src/lib/components/form-view-model/form-view-model.component.ts [hunk 6]
```diff
@@ -300,6 +328,7 @@ export class FormViewModelComponent implements OnInit {
               this.change$.pipe(take(1)).subscribe(() => {
                 this.loading$.next(false);
               });
+              this._isWizard = this.formio.form.display == 'wizard';
```
Comment (teodor-ritense) [inline projects/valtimo/form-view-model/src/lib/components/form-view-model/form-view-model.component.ts]:
```suggestion
              this._isWizard = this.formio.form.display === 'wizard';
```

File: projects/valtimo/form-view-model/src/lib/services/view-model.service.ts [hunk 7]
```diff
@@ -42,13 +42,20 @@ export class ViewModelService extends BaseApiService {
...
-      params: {
+    const params = {
         formName,
         taskInstanceId,
-      },
+        isWizard
+    }
+    if (!isNaN(page)) {
+      params['page'] = page;
+    }
```
Comment (teodor-ritense) [inline projects/valtimo/form-view-model/src/lib/services/view-model.service.ts]:
```suggestion
        isWizard,
        ...(!isNaN(page) && {page})
    }
```

File: projects/valtimo/form-view-model/src/lib/services/view-model.service.ts [hunk 8]
```diff
@@ -42,13 +42,20 @@ export class ViewModelService extends BaseApiService {
...
+    return this.httpClient.post(this.getApiUrl(`/v1/form/view-model/user-task`), viewModel, {
+      params: params,
```
Comment (teodor-ritense) [inline projects/valtimo/form-view-model/src/lib/services/view-model.service.ts]:
```suggestion
      params,
```

**Review by teodor-ritense — APPROVED** [review]

**Pull request merged**